src/tic-tac-toe.ts

```diff
@@ -40,36 +44,98 @@ class Game {
     return square === this.currentPlayer;
   };
 
-  private checkRowsForWin(): boolean {
-    return this.board.some(row => row.every(this.isCurrentPlayer));
+  private checkRowsForWin(): WinData {
+    const winData: WinData = {
+      hasWinner: false,
+      winningCells: [],
+    };
+
+    for (const rowIndex in this.board) {
+      if (this.board[rowIndex].every(cell => this.isCurrentPlayer(cell))) {
+        winData.hasWinner = true;
+        for (
+          let columnIndex = 0;
+          columnIndex < this.board.length;
+          columnIndex++
+        ) {
+          winData.winningCells.push([Number(rowIndex), columnIndex]);
+        }
+        break;
+      }
+    }
+
+    return winData;
   }
 
-  private checkColumnsForWin(): boolean {
+  private checkColumnsForWin(): WinData {
+    const winData: WinData = {
+      hasWinner: false,
+      winningCells: [],
+    };
+
     const transposedBoard = this.eachIndex().map(i =>
       this.board.map(row => row[i])
     );
 
-    return transposedBoard.some(row => row.every(this.isCurrentPlayer));
+    winData.hasWinner = transposedBoard.some(row =>
+      row.every(this.isCurrentPlayer)
+    );
+
+    if (winData.hasWinner) {
+      transposedBoard.forEach((row, rowIndex) => {
+        if (row.every(value => this.isCurrentPlayer(value))) {
```

Comment:
This is the same logic as the call to `some` above; so we first check _all_ columns to see if any is filled, and then check each column. This might be easier to read/reason about if this was all done in the same loop.